base/shell/explorer/notifyiconscust.cpp

```diff
@@ -20,31 +20,193 @@
 
 #include "precomp.h"
 
+#define BITMAP_WIDTH 16
+#define BITMAP_HEIGHT 16
+
+#define IDC_NOTIFY_OPTIONS 5000
+
+typedef struct
+{
+    DWORD hwnd;
+    DWORD uId;
+    DWORD uCallbackMsg;
+    DWORD unkow[2];
+    HICON hIcon;
+    WCHAR text[64];
+}TRAYDATA;
+
+// Returns TRUE if successful, and FALSE otherwise. 
+BOOL AddListViewColumn(HWND hWndListView, LPWSTR label, INT iCol, int w)
+{
+    LVCOLUMNW lvc;
+
+    lvc.mask = LVCF_FMT | LVCF_WIDTH | LVCF_TEXT | LVCF_SUBITEM;
+    lvc.iSubItem = iCol;
+    lvc.pszText = label;
+    lvc.cchTextMax = lstrlenW(label);
+    lvc.cx = w;
+    lvc.fmt = LVCFMT_CENTER;
+
+    return (ListView_InsertColumn(hWndListView, iCol, &lvc) != -1);
+}
+
+// Returns TRUE if successful, and FALSE otherwise. 
+BOOL InsertItem(HWND hListView, TBBUTTON *tb, TRAYDATA *data, int id)
+{
+    LVITEMW iItem;
+
+    iItem.mask = LVIF_TEXT | LVIF_IMAGE;
+    iItem.pszText = data->text;
+    iItem.cchTextMax = lstrlenW(data->text);
+    iItem.iSubItem = 0;
+    iItem.iItem = id;
+    iItem.iImage = id;
+
+    return (ListView_InsertItem(hListView, &iItem) != -1);
```

Comment:
with this new implementation, you're not able to raise a ERR() in case of failure (in particular since it's very unlikely)

Reply:
One doesn't need to feel concerned about displaying messages for unlikely errors for every line of code (It makes debug code much more larger, would pollute the debug log etc.), especially since this is not critical code whose failure could be dramatic (and hence where one would be genuinely interested to know why it failed).

Reply:
OK, good point.